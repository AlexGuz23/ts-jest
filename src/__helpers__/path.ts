import { resolve, relative } from 'path'

export const ROOT = resolve(__dirname, '..', '..')

export function relativeToRoot(path: string) {
  return relative(ROOT, path)
}
