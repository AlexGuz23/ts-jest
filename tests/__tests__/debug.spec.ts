import * as fs from 'fs-extra';
import * as path from 'path';
import runJest from '../__helpers__/runJest';

const debugFilePath = path.resolve(
  __dirname,
  '../simple/node_modules/ts-jest/debug.txt',
);

describe('Debug output', () => {
  beforeEach(async () => {
    return fs.remove(debugFilePath);
  });

  it('should create a debug file with the correct output if the flag is set', async () => {
    runJest('../simple', ['--no-cache', '-u'], {
      TS_JEST_DEBUG: 'true',
    });
    const logFile = await fs.readFile(debugFilePath, 'utf8');

    expect(logFile).not.toBeNull();
  });

  it('Should not create a file if the debug flag is not set', async () => {
    runJest('../simple', ['--no-cache', '-u']);
    expect.assertions(1); // To make sure we actually do assert the promise on the line below
    await expect(fs.readFile(debugFilePath, 'utf8')).rejects.toThrow();
  });
});
