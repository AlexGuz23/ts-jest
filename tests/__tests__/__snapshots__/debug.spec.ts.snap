// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Debug output should create a debug file with the correct output if the flag is set 1`] = `
"[
  \\"Reading tsconfig file from path tsconfig.json\\"
]
[
  \\"Original typescript config before modifications: \\",
  {
    \\"target\\": 1,
    \\"module\\": 1,
    \\"moduleResolution\\": 2,
    \\"noEmitOnError\\": false,
    \\"jsx\\": 2,
    \\"allowJs\\": true,
    \\"inlineSourceMap\\": true,
    \\"inlineSources\\": true
  }
]
[
  \\"Reading tsconfig file from path tsconfig.json\\"
]
[
  \\"final compilerOptions:\\",
  {
    \\"target\\": 1,
    \\"module\\": 1,
    \\"moduleResolution\\": 2,
    \\"noEmitOnError\\": false,
    \\"jsx\\": 2,
    \\"allowJs\\": true,
    \\"inlineSourceMap\\": true,
    \\"inlineSources\\": true
  }
]
[
  \\"tsJestConfig: \\",
  {}
]
[
  \\"Using babel with options:\\",
  {
    \\"babelrc\\": false,
    \\"plugins\\": [],
    \\"presets\\": []
  }
]"
`;
